designer/server/src/main/scala/pl/touk/nussknacker/ui/statistics/ScenarioStatistics.scala

```diff
@@ -0,0 +1,138 @@
+package pl.touk.nussknacker.ui.statistics
+
+import cats.implicits.{toFoldableOps, toTraverseOps}
+import pl.touk.nussknacker.engine.api.component.ComponentType
+import pl.touk.nussknacker.restmodel.component
+import pl.touk.nussknacker.ui.process.repository.DbProcessActivityRepository
+
+import java.time.Instant
+
+object ScenarioStatistics {
+
+  def getGeneralStatistics(scenariosInputData: List[ScenarioStatisticsInputData]): Map[String, String] = {
+    if (scenariosInputData.isEmpty) {
+      Map.empty
+    } else {
+      val scenariosCount = scenariosInputData.count(!_.isFragment)
+      //        Nodes stats
+      val sortedNodes  = scenariosInputData.map(_.nodesCount).sorted
+      val nodesMedian  = sortedNodes.get(sortedNodes.length / 2).getOrElse(0)
+      val nodesAverage = sortedNodes.sum / scenariosInputData.length
+      val nodesMax     = sortedNodes.head
+      val nodesMin     = sortedNodes.last
+      //        Category stats
+      val categoriesCount = scenariosInputData.map(_.scenarioCategory).toSet.size
+      //        Version stats
+      val sortedVersions  = scenariosInputData.map(_.scenarioVersion.value).sorted
+      val versionsMedian  = sortedVersions.get(sortedVersions.length / 2).getOrElse(0)
+      val versionsAverage = sortedVersions.sum / scenariosInputData.length
+      val versionsMax     = sortedVersions.head
+      val versionsMin     = sortedVersions.last
+      //        Author stats
+      val authorsCount = scenariosInputData.map(_.createdBy).toSet.size
+      //        Fragment stats
+      val fragments        = scenariosInputData.map(_.fragmentsUsedCount).sorted
+      val fragmentsMedian  = fragments.get(fragments.length / 2).getOrElse(0)
+      val fragmentsAverage = fragments.sum / scenariosCount
+      //          Uptime stats
+      val lastActions = scenariosInputData.map(_.lastDeployedAction).filter(_.isDefined).sequence.get
```

Comment:
get on Option is unsafe, maybe we could use: `toList.flatten` or map actions to uptime and then use getOrElse?